### PR TITLE
Improve history support

### PR DESCRIPTION
**Pull request by jdesboeufs**
- `MOD=990` are now marked as `ERRORED` and not exported anymore
- `MOD=321` (simple fusion) is now supported
- `MOD=320` (simple absorption) is now correctly supported (i.e Tours, Digne…)
- `MOD=300` (deletion) is now supported: update of END_DATE

For deletion, we could introduce a new direction symbol: `-x`.
WDYT?

Feel free to intensively review this PR, I've no significative experience in Python ;)

### CODE REVIEW

**Comment by jdesboeufs** [comment]
```
Geohisto: 35891 actual INSEE codes
COG: 35885 actual INSEE codes
Towns in geohisto but not in COG 2016:
14624
27255
49382
52427
55273
68031
95025
95065
Towns in COG 2016 but not in geohisto:
14472
64235
```

File: utils.py [hunk 1]
```diff
@@ -60,16 +60,30 @@ def has_been_restablished(town, history):
 
 def has_been_renamed(town, history):
     """Return `True` in case of a rename with the same INSEE code."""
-    return (int(history['MOD']) in (120, 331)
+    return (int(history['MOD']) in (120, 311, 331)
             and history['NCCOFF'] != town[0]['NCCENR'])
 
+def has_been_deleted(town, history):
+    """Return `True` in case of a deletion."""
+    return (int(history['MOD']) == 300)
```
Comment (davidbgk) [inline utils.py]: External parenthesis are useless in this case.

File: utils.py [hunk 2]
```diff
@@ -60,16 +60,30 @@ def has_been_restablished(town, history):
...
+
+
+def has_errored_numerotation(town, history):
+    """Return `True` in case of a errored numerotation."""
+    return (int(history['MOD']) == 990)
```
Comment (davidbgk) [inline utils.py]: Same here.

File: utils.py [hunk 3]
```diff
@@ -60,16 +60,30 @@ def has_been_restablished(town, history):
...
 def has_ancestor(town, towns, history):
     """Return `True` in case of a merge with a different name."""
-    return (int(history['MOD']) in (320, 340, 341)
+    return (int(history['MOD']) in (340, 321, 341)
             and town[0]['NCCENR'] != towns[history['COMECH']][0]['NCCENR'])
 
 
+def has_absorbed(town, history):
+    """Return `True` in case of a simple town absorption."""
+    return (int(history['MOD']) == 320)
```
Comment (davidbgk) [inline utils.py]: And here.

**Pull request merged**